Gems/Multiplayer/Code/CMakeLists.txt

```diff
@@ -35,6 +34,89 @@ ly_add_target(
         *.AutoPackets.xml,AutoPackets_Source.jinja,$path/$fileprefix.AutoPackets.cpp
         *.AutoPackets.xml,AutoPacketDispatcher_Header.jinja,$path/$fileprefix.AutoPacketDispatcher.h
         *.AutoPackets.xml,AutoPacketDispatcher_Inline.jinja,$path/$fileprefix.AutoPacketDispatcher.inl
+)
+
+ly_add_target(
+    NAME Multiplayer.Client.Static STATIC
+    NAMESPACE Gem
+    FILES_CMAKE
+        multiplayer_files.cmake
```

Comment:
This multiplayer_files.cmake needs to be as small as possible.
We want to avoid compiling the same file with multiple different compiler flags as much as possible.

Actually per PAL we should use separate files instead of macros

Reply:
I can examine the file sets to optimize it further. This isn't a very complete attempt in that regard.

There's definitely reduction that can happen here. The pro of the AzAutoGen changes here is an alternate ruleset specific to each target type can be generated. Though that to me seems functionally equivalent to macro usage with the added maintenance cost of jinja variations. Is there a tradeoff here outside of preventing misuse of the macros? 

Further downstream, I want to avoid exploding out Multiplayer as much as possible. I took a look at separating out into distinct files initially and the Unified requirement makes it challenging.

Reply:
I don't think different files are going to be viable here. We really need the flexibility to go from client to server to unified to hit all our customer use-cases; and we need to do so without convoluting our class architecture and making it harder to author multiplayer content than it already is. Refer to the original discussion we had between you and me and Esteban and a few others.